### PR TITLE
Eob totals

### PR DESCRIPTION
**Pull request by juggernautsei**
Total up the charges and balance columns of the EOB page

eob_totals

### CODE REVIEW

File: library/globals.inc.php [hunk 1]
```diff
@@ -1866,7 +1866,12 @@
       '0',
       xl('Enable Onsite Patient Portal.')
     ),
-
+    'portal_statements' => array(
```
Comment (bradymiller) [inline library/globals.inc.php]: Where are you using this global?

File: interface/billing/sl_eob_search.php [hunk 2]
```diff
@@ -1071,7 +1147,19 @@ function npopup(pid) {
 
 if (!$INTEGRATED_AR) SLClose();
 ?>
-
+<tr bgcolor='<?php echo $bgcolor ?>'>
+<td>Totals</td>
```
Comment (bradymiller) [inline interface/billing/sl_eob_search.php]: Need to translate Totals

File: interface/billing/sl_eob_search.php [hunk 3]
```diff
@@ -462,6 +462,82 @@ function npopup(pid) {
...
+         result += thisNumber;
+	 } // end for
+		 
+  } // end try
+  catch (ex)
+  {
+     //window.alert("Exception in function computeTableColumnTotal()\n" + ex);
+     result = 0;
+  }
+  finally
```
Comment (bradymiller) [inline interface/billing/sl_eob_search.php]: finally will only work with PHP 5.5+ . Is there a way to do this without needing the fonally command?

File: library/globals.inc.php (hunk 1 above)
Comment (juggernautsei) [inline library/globals.inc.php]: This feature is in this branch reviewed by Art
https://github.com/juggernautsei/openemr/tree/portal_invoice

Thread is here
[Patient Portal Statements](https://sourceforge.net/p/openemr/discussion/202506/thread/ce222227/?limit=25#8200)

File: interface/billing/sl_eob_search.php (hunk 3 above)
Comment (juggernautsei) [inline interface/billing/sl_eob_search.php]: This finally is not for PHP to my knowledge. It is in the javascript section. The javascript will execute with or without the "finally" at the end. The finally statement is there to execute regardless of the result of the preceding code.

File: interface/billing/sl_eob_search.php (hunk 2 above)
Comment (juggernautsei) [inline interface/billing/sl_eob_search.php]: I will add the xl() to translate the totals.

File: interface/billing/sl_eob_search.php (hunk 3 above)
Comment (bradymiller) [inline interface/billing/sl_eob_search.php]: oops. disregard my above comment then :)

File: interface/billing/sl_eob_search.php (hunk 2 above)
Comment (bradymiller) [inline interface/billing/sl_eob_search.php]: use xlt() function:
``````
<td><?php echo xlt('Totals'); ?></td>
``````

File: library/globals.inc.php (hunk 1 above)
Comment (bradymiller) [inline library/globals.inc.php]: I would then remove this global since it is unrelated to the eob totals feature.

Comment (juggernautsei) [inline library/globals.inc.php]: I have removed this globals from this entry.

File: interface/billing/sl_eob_search.php (hunk 2 above)
Comment (juggernautsei) [inline interface/billing/sl_eob_search.php]: Brady,
I have been going over and over this request. The translation would have to take place inside the JavaScript at line 502. Note that variable is placed by the JavaScript and not the PHP.  So is there an xlt for JavaScript?

Comment (bradymiller) [inline interface/billing/sl_eob_search.php]: Hi Sherwin,
But I'm not asking to translate the variables, just the 'Totals' string.
-brady

Comment (juggernautsei) [inline interface/billing/sl_eob_search.php]: https://github.com/juggernautsei/openemr/blob/7e6ca38d3bc71265c0b85e798a2f54a7b4a32ff6/interface/billing/sl_eob_search.php

I have made the addition the file.

Sherwin
757-328-2736
Open Med Practice

“For those who believe, no proof is necessary. For those who don't believe,
no proof is possible.” ― Stuart Chase
<http://www.goodreads.com/author/show/1051789.Stuart_Chase>

**Comment by bradymiller** [comment]
Hi rebased your code into 1 commit to test it out:
https://github.com/bradymiller/openemr/commit/9fbbb3667bd53caa921c39137ab9d4a577598e42

And it does not work(Only the Totals label shows up; it is not totaling the numbers)

**Pull request closed without merging**